lib/core/CLogger.cc

```diff
@@ -6,61 +6,121 @@
-#include <sstream>
 #include <stdexcept>
 
-#include <errno.h>
-#include <stdlib.h>
-#include <string.h>
-
 // environ is a global variable from the C runtime library
 #ifdef Windows
 __declspec(dllimport)
 #endif
     extern char** environ;
 
 namespace {
+const std::string TRACE{"TRACE"};
+const std::string DEBUG{"DEBUG"};
+const std::string INFO{"INFO"};
+const std::string WARN{"WARN"};
+const std::string ERROR{"ERROR"};
+const std::string FATAL{"FATAL"};
+
 // To ensure the singleton is constructed before multiple threads may require it
 // call instance() during the static initialisation phase of the program.  Of
 // course, the instance may already be constructed before this if another static
 // object has used it.
 const ml::core::CLogger& DO_NOT_USE_THIS_VARIABLE = ml::core::CLogger::instance();
 
-std::mutex fatalErrorHandlerMutex;
+// These must use boost::shared_ptr, not std, as that's what the Boost.Log interface
+// uses
+using TOStreamPtr = boost::shared_ptr<std::ostream>;
+using TTextOStream = boost::log::sinks::text_ostream_backend;
+using TTextOStreamSynchronousSink = boost::log::sinks::synchronous_sink<TTextOStream>;
+
+class CTimeStampFormatterFactory
+    : public boost::log::basic_formatter_factory<char, boost::posix_time::ptime> {
+private:
+    static const std::string FORMAT;
+
+public:
+    formatter_type create_formatter(const boost::log::attribute_name& name,
+                                    const args_map& args) {
+        args_map::const_iterator iter{args.find(FORMAT)};
+        if (iter != args.end()) {
+            return boost::log::expressions::stream
+                   << boost::log::expressions::format_date_time<boost::posix_time::ptime>(
+                          boost::log::expressions::attr<boost::posix_time::ptime>(name),
+                          iter->second);
+        } else {
+            return boost::log::expressions::stream
+                   << boost::log::expressions::attr<boost::posix_time::ptime>(name);
+        }
+    }
+};
+
+const std::string CTimeStampFormatterFactory::FORMAT{"format"};
 }
 
 namespace ml {
 namespace core {
 
 CLogger::CLogger()
-    : m_Logger(0), m_Reconfigured(false), m_ProgramName(CProgName::progName()),
+    : m_Reconfigured(false), m_FileAttributeName("File"),
+      m_LineAttributeName("Line"), m_FunctionAttributeName("Function"),
       m_OrigStderrFd(-1), m_FatalErrorHandler(defaultFatalErrorHandler) {
     CCrashHandler::installCrashHandler();
+    // These formatter factories are not needed for programmatic configuration
+    // of the format, but without them formats defined in settings files don't
+    // work correctly.
+    boost::log::register_simple_formatter_factory<ELevel, char>("Severity");
+    boost::log::register_simple_filter_factory<ELevel, char>("Severity");
+    boost::log::register_formatter_factory(
+        "TimeStamp", boost::make_shared<CTimeStampFormatterFactory>());
+    auto loggingCorePtr{boost::log::core::get()};
```

Comment:
With auto I always use copy initialisation to avoid having to think about the `auto-deduced-a-std::initializer_list` problem. Maybe stick to that convention in this file?
```suggestion
    auto loggingCorePtr = boost::log::core::get();
```